ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala

```diff
@@ -218,13 +231,111 @@ private class JdbcLedgerDao(
+              // pattern as with transactions.
+              Some(
+                s"Generation mismatch: expected=$expGeneration, actual=${configuration.generation}"
+              )
+
+            case _ =>
+              // Rejection reason was set, or we have no previous configuration generation, in which case we accept any
+              // generation.
+              rejectionReason
+          }
+
+        val update = finalRejectionReason match {
+          case None =>
+            Update.ConfigurationChanged(
+              recordTime = Time.Timestamp.assertFromInstant(recordedAt),
+              submissionId =
+                submissionId.asInstanceOf[SubmissionId], // TODO append-only: FIXME type cast
+              participantId = null.asInstanceOf[
+                v1.ParticipantId
+              ], // TODO append-only: FIXME type cast // not used for DBDTO generation
+              newConfiguration = configuration,
+            )
+
+          case Some(reason) =>
+            Update.ConfigurationChangeRejected(
+              recordTime = Time.Timestamp.assertFromInstant(recordedAt),
+              submissionId =
+                submissionId.asInstanceOf[SubmissionId], // TODO append-only: FIXME type cast
+              participantId = null.asInstanceOf[
+                v1.ParticipantId
+              ], // TODO append-only: FIXME type cast // not used for DBDTO generation
+              proposedConfiguration = configuration,
+              rejectionReason = reason,
+            )
+        }
+        sequentialIndexer.store(conn, offsetStep.offset, Some(update))
+        PersistenceResponse.Ok
+      }
+    }
 
   override def storePartyEntry(
       offsetStep: OffsetStep,
       partyEntry: PartyLedgerEntry,
-  )(implicit loggingContext: LoggingContext): Future[PersistenceResponse] =
-    throw new UnsupportedOperationException("not supported") // TODO add support
+  )(implicit loggingContext: LoggingContext): Future[PersistenceResponse] = {
+    logger.info("Storing party entry")
+    dbDispatcher.executeSql(metrics.daml.index.db.storePartyEntryDbMetrics) { implicit conn =>
+      val savepoint = conn.setSavepoint()
+      partyEntry match {
+        case PartyLedgerEntry.AllocationAccepted(submissionIdOpt, recordTime, partyDetails) =>
+          Try({
+            sequentialIndexer.store(
+              conn,
+              offsetStep.offset,
+              Some(
+                Update.PartyAddedToParticipant(
+                  party = partyDetails.party,
+                  displayName = partyDetails.displayName.orNull,
+                  participantId = participantId,
+                  recordTime = Time.Timestamp.assertFromInstant(recordTime),
+                  submissionId = submissionIdOpt,
+                )
+              ),
+            )
+            PersistenceResponse.Ok
+          }).recover {
+            case NonFatal(e) if e.getMessage.contains(queries.DUPLICATE_KEY_ERROR) =>
```

Comment:
In `storeConfigurationEntry`, you have removed the savepoint-rollback-on-conflict in favor of an idempotent insert. Can we do the same here?

Reply:
We most certainly can, but that is also a gray area to me: let's try to understand the whole picture in a discussion jointly with the configuration changes.

Reply:
as discussed we agreed to not change this to keep implementation as close as possible to the original. in that effort we agreed to use this approach here with the configuration change as well (which also gives us back the warning logs as well)